### ISSUE TITLE
Athena - Make query execution metadata available

### ISSUE THREAD

**Issue by flaviomax** [awslabs/aws-data-wrangler#312]
**Is your feature request related to a problem? Please describe.**
When using the athena module's `read_sql_query`, all we get is the result. However, for some debugging and cost monitoring purposes, having the query metadata (e.g. query_id) can also be useful. 
A possible use case I've already been through is when calling AWS support -- they always request the query ID.

**Describe the solution you'd like**
Ideally, there should be a way to access the `last_query_metadata` in a "Wrangler Athena" object. That would mean a huge refactor for the lib, though :/
In the current structure, maybe we could add a boolean parameter to the query which would also return the query metadata, not just the results. But that can be a little confusing, regarding typing

**Comment by igorborgest** [comment]
@flaviomax Yeah, this is not the first time I've heard about this requirement. Unfortunately seems that there is not easy/elegant way to do it.

An alternative would be the injection of this metadata as attributes into the Pandas DataFrame.

*e.g.*
```py3
df = wr.athena.read_sql_query()
print(df.metadata.data_scanned)
```

Others possibles values to expose ([ref](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/athena.html#Athena.Client.get_query_execution)):
```py3
'Statistics': {
    'EngineExecutionTimeInMillis': 123,
    'DataScannedInBytes': 123,
    'DataManifestLocation': 'string',
    'TotalExecutionTimeInMillis': 123,
    'QueryQueueTimeInMillis': 123,
    'QueryPlanningTimeInMillis': 123,
    'ServiceProcessingTimeInMillis': 123
}
```

What do you think?

**Comment by flaviomax** [comment]
I think your solution sounds pretty elegant! I did not know you could freely add meta-information to a DataFrame object (as described [here](https://stackoverflow.com/questions/14688306/adding-meta-information-metadata-to-pandas-dataframe), too)

I think a nice solution would be to add an attribute to the `df` returned by `read_sql_query` (and `read_sql_table`, by extension) with the whole `QueryExecution` object returned by boto's `get_query_execution`, instead of just the Statistics you mentioned. What do you think?

If you don't mind, I could implement this :)

**Comment by igorborgest** [comment]
We should return as much metadata as possible indeed, so I like the idea to return the whole `QueryExecution` object.

For sure would me another cool contribution, let me assign you on it.

**Comment by igorborgest** [comment]
Released in [1.7.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.7.0)!
